components/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/TenantService.java

```diff
@@ -187,21 +179,13 @@ public Response delete(@ApiParam(value = "Tenant name", required = true) @PathPa
     private TenancyProvider getTenancyProvider() throws DeploymentEnvironmentException {
         String platform = System.getenv(ENV_DEPLOYMENT_PLATFORM);
         if (platform == null || platform.equals("")) {
-            throw new DeploymentEnvironmentException("Unable to identify the deployment platform.");
+            throw new DeploymentEnvironmentException("Unable to identify the deployment platform");
         }
 
-        if (platform.toLowerCase().equals(DEPLOYMENT_KUBERNETES)) {
+        if (platform.equalsIgnoreCase(DEPLOYMENT_KUBERNETES)) {
```

Comment:
Should not this be checked in the other way around to avoid NPEs ? We can anyway consider it as a best practice.